internal/engine/compiler/compiler_value_location.go

```diff
@@ -201,10 +201,10 @@ func (v *runtimeValueLocationStack) push(loc *runtimeValueLocation) {
 	} else {
 		v.stack[v.sp] = loc
 	}
+	v.sp++
```

Comment:
so stackPointerCeil must be set after sp is incremented :D

Reply:
as we skip the stack grow if `len(valueStack) >=  basePointer+stackPointerCeil`.

Reply:
sounds like a comment for the code :D

Reply:
done